eldritch/events.py

```diff
@@ -565,10 +566,16 @@ def resolve(self, state):
     if self.drawn is None:
       self.drawn = []
       deck = getattr(state, self.deck)
-      for _ in range(self.draw_count):
-        if not deck:
+      i = 0
+      while len(self.drawn) < self.draw_count:
+        i += 1
+        if not deck or i > 100:
```

Comment:
why the i > 100?  Is this just arbitrary?

Reply:
Yep. In the unlikely event that there are no weapons remaining, it should stop at some point.

Reply:
seems like we should be able to get a deck length (found before the start of the loop) and use that instead.  Will make it more robust to adding additional cards to the deck.  There should also be a test for when none of the item type exist.